api/src/db/models/opportunity_models.py

```diff
@@ -127,12 +120,39 @@ class OpportunitySummary(Base, TimestampMixin):
     agency_email_address: Mapped[str | None]
     agency_email_address_description: Mapped[str | None]
 
+    is_deleted: Mapped[bool | None]
+
     can_send_mail: Mapped[bool | None]
     publisher_profile_id: Mapped[int | None]
     publisher_user_id: Mapped[str | None]
     updated_by: Mapped[str | None]
     created_by: Mapped[str | None]
 
+    link_funding_instruments: Mapped[list["LinkFundingInstrumentSummary"]] = relationship(
+        back_populates="opportunity_summary", uselist=True, cascade="all, delete-orphan"
+    )
+    link_funding_categories: Mapped[list["LinkFundingCategorySummary"]] = relationship(
+        back_populates="opportunity_summary", uselist=True, cascade="all, delete-orphan"
+    )
+    link_applicant_types: Mapped[list["LinkApplicantTypeSummary"]] = relationship(
+        back_populates="opportunity_summary", uselist=True, cascade="all, delete-orphan"
+    )
+
+    @property
+    def funding_instruments(self) -> list[FundingInstrument]:
+        # Helper method for serialization of the API response
+        return [f.funding_instrument for f in self.link_funding_instruments]
```

Comment:
Would it make sense to return a set instead of a list for this one and the two below? I'm assuming they are unordered and have no repeats.

```suggestion
    def funding_instruments(self) -> set[FundingInstrument]:
        # Helper method for serialization of the API response
        return {f.funding_instrument for f in self.link_funding_instruments}
```

Reply:
Yeah, you're right. The nature of the relationship in the DB would mean they're unique (the link table primary key is the summary ID + the lookup code). So I should be able to make it a set